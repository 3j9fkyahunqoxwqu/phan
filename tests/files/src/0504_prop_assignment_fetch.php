<?php

class Test504 {
    /** @var int[] */
    public static $prop = [2];
    /** @var int[] */
    public static $otherProp = [3];

    public static function test() {
        $t = self::class;
        $t::$prop = 2;
        echo $t::$otherProp;
        $t::$propMissing = 2;
        echo $t::$otherMissingProp;
    }
}
