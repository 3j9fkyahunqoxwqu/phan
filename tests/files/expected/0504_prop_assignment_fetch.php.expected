%s:11 PhanTypeMismatchProperty Assigning 2 to property but \Test504::prop is array{0:2}|int[]
%s:12 PhanTypeConversionFromArray array to string conversion
%s:13 PhanUndeclaredStaticProperty Static property 'propMissing' on \Test504 is undeclared
%s:14 PhanUndeclaredStaticProperty Static property 'otherMissingProp' on \Test504 is undeclared
